polar-core/src/data_filtering.rs

```diff
@@ -167,35 +165,14 @@ impl From<VarInfo> for Vars {
             .collect::<HashMap<_, _>>();
 
         let fields = info.field_relationships;
-        let triples = {
-            let mut assign_id = |item: &Symbol| get_var_id(&mut variables, item.clone(), &counter);
-            fields
-                .iter()
-                .map(|(p, f, c)| (assign_id(p), f, assign_id(c)))
-                .collect::<Vec<_>>()
-        };
-
-        // if x.y=a and x.y=b then a=b, so add a cycle.
```

Comment:
we don't need to do this anymore because `dot_var` avoids creating redundant variables